diff --git a/dquote.c b/dquote.c
index e2c03f64eb3..2c45b7a2701 100644
--- a/dquote.c
+++ b/dquote.c
@@ -58,7 +58,7 @@ Perl_grok_bslash_c(pTHX_ const char   source,
     if (isPRINT_A(*result) && ckWARN(WARN_SYNTAX)) {
         U8 clearer[3];
         U8 i = 0;
-        char format[] = PERL_DIAG_WARN_SYNTAX("\"\\c%c\" is more clearly written simply as \"%s\"");
+        char format[] = "\"\\c%c\" is more clearly written simply as \"%s\"";
 
         if (! isWORDCHAR(*result)) {
             clearer[i++] = '\\';
